src/lib/ext-listen.js

```diff
@@ -0,0 +1,29 @@
+export async function getAllExtensions() {
+  const extensions = await browser.management.getAll();
+  return extensions.filter((extension) => extension.type !== 'theme');
+}
+
+export async function areExtsBeingMonitored() {
+  const { status } = await browser.runtime.sendMessage({
+    getMonitoringStatus: true,
+  });
+  return status;
+}
+
+export async function initMonitorAll(extensions) {
+  return await browser.runtime.sendMessage({
+    extStartMonitorAllExts: extensions,
+  });
+}
+
+export async function stopMonitorAll() {
+  return await browser.runtime.sendMessage({
+    extStopMonitorAll: true,
+  });
+}
+
+export function viewActivityLogs() {
+  browser.tabs.create({
+    url: `${browser.runtime.getURL('../activitylog/activitylog.html')}`,
```

Comment:
In ext-monitor.js there is a similar `runtime.getURL` (in the `isExtensionPageOpen` method) but with a different url, I'm pretty sure that the urls are different because they are both relative urls and the callers are at a different nesting level in the extension's directory tree. I suggest to use an absolute url (I mean `browser.runtime.getURL('/activitylog/activitylog.html')` and possibly to do it in a single place, e.g. a static method or a function exported by a module used by the other js modules that needs the shared helper) to make it easier to be sure that the url would be resolved to the expected one

Reply:
Thanks for noticing, I have added a function named `extPageURL()` in `ext-listen.js` with absolute url. I resued this function two times.